tests/test_combine_sregions.py

```diff
@@ -0,0 +1,418 @@
+    expected = "POLYGON ICRS  1.000000000 0.000000000 0.000000000 0.000000000 0.000000000 1.000000000 0.500000000 1.000000000 0.500000000 1.500000000 1.500000000 1.500000000 1.500000000 0.500000000 1.000000000 0.500000000"
+    assert combined_sregion == expected
+
+
+def test_sregion_complex(complex_footprint_set, det2world):
+    """Test for a complex set of footprints as S_REGION strings."""
+    sregion_list = _footprints_to_sregion_list(complex_footprint_set)
+    combined_sregion = combine_sregions(sregion_list, det2world)
+
+    assert combined_sregion.count("POLYGON ICRS") == 3
+    assert combined_sregion.startswith("POLYGON ICRS")
+    sregions_out = [s.strip() for s in combined_sregion.split("POLYGON ICRS")][1:]
+    footprints_out = [sregion_to_footprint(sregion) for sregion in sregions_out]
+
+    # Too tedious to check exact coordinates, so just check the shapes of the regions.
+    region_shapes = sorted([region.shape for region in footprints_out])
+    expected_shapes = [(4, 2), (16, 2), (20, 2)]
+    assert region_shapes == expected_shapes
+
+
+def test_sregion_intersection(complex_footprint_set, det2world):
+    """
+    Test for a complex set of footprints intersected with a bounding box.
+
+    The bounding box is chosen to intersect only two of the three combined regions,
+    and to cut both of them.
+    """
+    bbox = np.array(
+        [
+            [178.5, 0.5],
+            [181.5, 0.5],
+            [181.5, 1.5],
+            [178.5, 1.5],
+        ]
+    )
+    sregion_list = _footprints_to_sregion_list(complex_footprint_set)
+    combined_sregion = combine_sregions(sregion_list, det2world, intersect_footprint=bbox)
+
+    assert combined_sregion.count("POLYGON ICRS") == 2
+    assert combined_sregion.startswith("POLYGON ICRS")
+    expected = "POLYGON ICRS  179.000000000 1.000000000 179.000000000 0.500000000 178.500000000 0.500000000 178.500000000 1.000000000  POLYGON ICRS  180.500000000 1.100000000 180.500000000 1.200000000 180.900000000 1.200000000 180.900000000 1.400000000 181.000000000 1.400000000 181.000000000 1.500000000 181.500000000 1.500000000 181.500000000 0.500000000 180.000000000 0.500000000 180.000000000 1.000000000 180.100000000 1.000000000 180.100000000 1.100000000"
+    assert combined_sregion == expected
+
+
+def test_sregion_no_overlap(complex_footprint_set, det2world):
+    """Test error raise when intersection footprint does not overlap any input footprints."""
+    bbox = np.array(
+        [
+            [10.0, 10.0],
+            [11.0, 10.0],
+            [11.0, 11.0],
+            [10.0, 11.0],
+        ]
+    )
+    sregion_list = _footprints_to_sregion_list(complex_footprint_set)
+    with pytest.raises(ValueError) as excinfo:
+        combine_sregions(sregion_list, det2world, intersect_footprint=bbox)
+    assert str(excinfo.value) == "No overlap between input s_regions and intersection footprint"
+
+
+def test_duplicate_tolerance():
```

Comment:
@tapastro This test is to ensure that we follow the tolerance requirements for duplicated points set by MAST.

RE your comment about repeatability of testing, I agree but I think it's fine to basically make any choice as long as it's deterministic.  The way the code is currently written, it will always just keep the first point in the polygon.  The Shapely union orders the points clockwise around the polygon.